config/config.yaml

```diff
@@ -292,23 +298,6 @@ tide:
     - do-not-merge/cherry-pick-not-approved
     - do-not-merge/work-in-progress
     - needs-rebase
-  # Maintain separate cert-manager configuration to make it easy to enable code freeze
```

Comment:
Have we not been freezing merges during a codefreeze near release? If so, +1 to this going. But this was here so that we could block merges unless they were explicitly in the milestone we are soon to be releasing.

(i.e. near 1.7 release, PRs *must* have the `v1.7` milestone on them to be merged to master). The idea being to make it possible to lgtm/approve stuff despite it being targeted to land in the `v1.8` release.

Reply:
That's not the documented process anyway. https://cert-manager.io/docs/contributing/release-process/#process-for-releasing-a-version

We fast-forward the release-x.y.z branch for alpha releases  and for the first beta release
and then do cherry-picks after that.

What's the use-case for / advantage of the milestone check?
I guess if there's a feature  which is definitely not to be in the next release and that is being merged in parts and partly merged even before the first beta, then that should be merged not into master, but into a feature-branch which can then be merged once the feature is complete.

Reply:
The idea stems from the "code freeze" concept in Kubernetes where after a certain point in the release (I believe when the first beta is cut), "milestone maintainers" are the only ones that can use the `/milestone` command to add a milestone to PRs.

The master branch begins requiring that PRs into master *must* be explicitly approved for the upcoming release - so anything that isn't approved won't actually merge until the code freeze is lifted (which I believe in k8s is when the first *RC* of the next release is cut).

This means:

* during the 'alpha' phase (i.e. immediately after release N-1), PRs targeting master must have `lgtm` and `approved` only (like we do today)
* during the 'beta' phase (i.e. immediately after the first *beta* of release N), PRs targeting master must have `lgtm`, `approved` *and* target milestone `N`. We then continue to *fast forward* the release branch to the tip of HEAD. The idea is to encourage PRs that focus on stability of the upcoming release (or in some instances, feature PRs that have been excepted as they are almost ready and just need a few extra fixes/deemed to be critical and there's been a discussion to allow it).
* during the 'rc' phase (i.e. immediately after the first *RC* release of N), PRs must be *cherry picked* into the appropriate release branch as 'master' is now tracking release N+1.

After the stable 'N' is cut, not much changes from the RC phase except things targeting the release branch are now going to land in `vMAJOR.N.1` (i.e. the first *patch* release of release N after the initial stable).

We don't do release candidates (RCs) so that step hasn't really ever been done, but we've enforced this for a handful of releases in the past (though not recently!)